### ISSUE TITLE
Add Aeotec Smart Switch 7

### ISSUE THREAD

**Issue by AlCalzone** [zwave-js/zwave-js#2634]
This seems premature until the config from @blhoward2 is available publicly as it seems a few people are tracking using this PR.

_Originally posted by @Nixon506E in https://github.com/zwave-js/node-zwave-js/issues/2027#issuecomment-842668658_

**Comment by tommyjlong** [comment]
HI, Wanted to get a status on this.....should I go ahead and try to make my own device file for this device until the official one comes out?

**Comment by blhoward2** [comment]
You’re welcome to do so. I only have a few left to do but I’ve had a busy few weeks at work.

**Comment by tommyjlong** [comment]
I'm working on a config device file and have a question.....  For the ZWA023 there is a config parameter: "Night Light On Time". It is a 4 byte value and the format is: `0xRRHHMMSS`  with the default of `0x00180000` which the manual says `means 6pm` (i.e 18 based on a 24 hour clock).

Focusing on the hour HH, since 0x18 means 18 decimal, this indicates to me that they are using Binary Encoded Decimal, whereby a nibble is used for each digit.  Regarding the device's config json file for this particular parameter, I understand that numbers are represented only as decimal (correct me if wrong), then I believe for HH, the values should be: `"defaultValue": 24` (i.e. x18), and the range is `"minValue": 0` to `"maxValue": 35` (x23). Is this correct?

**Comment by blhoward2** [comment]
While that may be right if you take the hex literally, I believe they use 18 as in the 18th hour. So the range is 0-23 for the HH. At least that is how I've always understood it.

I'll verify with Aeotec.

**Comment by blhoward2** [comment]
I confirmed with Aeotec. 0x018 means 6 pm. 0x019 means 7 pm, etc. @AlCalzone what does that do to the values?

**Comment by AlCalzone** [comment]
What the hell? So they actually have 0x00, 0x01, ..., 0x09, 0x10, 0x11, ..., 0x19, 0x20, 0x21, ..., 0x23?
Then we need to use `"options"` to model this and a range of 0..35.

**Comment by blhoward2** [comment]
Which means 59 options for the minute field. Unless we dumb it down instead of bitmasking it and give people one field with options every 30 minutes from 0000 to 2400 or something.

**Comment by AlCalzone** [comment]
That could be an option, but I'm sure that someone will want to set 19:57 or something :)

**Comment by blhoward2** [comment]
I'm not sure how we handled this elsewhere. I sort of think we left it manual entry and said see manual.

**Comment by AlCalzone** [comment]
honestly I'd be fine with that

**Comment by blhoward2** [comment]
Well, since its human readable in hex I vote that we let it manual entry and just put the format in the description. 235900 = 23:59 and 0 seconds.

Wait: That still wouldn’t work, right? because we’d convert the 235900 to hex?

**Comment by blhoward2** [comment]
See my edit, does that work? Aren’t we going to convert it to hex before we send it?

**Comment by blhoward2** [comment]
Maybe we need an option that sends the value through as hex instead of converting it?

**Comment by AlCalzone** [comment]
Users would have to convert `0x235900` to decimal before entering it. Maybe @robertsLando could add something to the UI that does that when users enter a `0x` prefixed hex number.

**Comment by tommyjlong** [comment]
As an FYI.....
 after running a few tests on my Smart Switch 7 I can confirm that the device does as Aeotec says and as the manual shows (i.e 0x18 is 6pm).

**Comment by blhoward2** [comment]
@robertsLando Can you do something similar to HA so we can make the description reference the hex value? Essentially if a hex value is put into a manual entry it should be treated as a hex and not a decimal.

**Comment by tommyjlong** [comment]
As a reference/FYI, here is a link to the device file I came up with:
https://github.com/tommyjlong/temp/blob/main/zwa023.json

**Comment by blhoward2** [comment]
Thanks. I made good progress on these last night so hope to push the official one tonight or tomorrow.
